clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -1410,6 +1410,14 @@ void SYCLToolChain::AddImpliedTargetArgs(const llvm::Triple &Triple,
                                           ":" + BackendOptName));
     }
   }
+  // only pass -vpfp-relaxed for aoc with -fintelfpga and fp-mode=fast
+  if (Args.hasArg(options::OPT_fintelfpga)) {
```

Comment:
This doesn't look to limit the option for `aoc` only.  an additional check will be needed to be sure we are targeting hardware.

Reply:
Thanks for the review Mike. Does this check ".IsFPGAHWMode()" works?